sonar-update-center-common/src/main/java/org/sonar/updatecenter/common/Release.java

```diff
@@ -370,8 +370,9 @@ public boolean equals(Object o) {
     return artifact.equals(release.artifact) && version.equals(release.version) && Objects.equals(product, release.product);
   }
 
-  public void setProduct(Product product) {
+  public Release setProduct(Product product) {
```

Comment:
I am not getting the dual behavior in this class with the setters. Some are void, and some return the Release object. Plus, they are scattered all over the place, mixed with other methods.
I know that it is not related to these code changes, but still worth mentioning! :D

Reply:
Right.... I think this behavior is actually encouraged unfortunately by SonarQube (or IntelliJ linter?) that marks code on yellow when you return something from the method and "the return value is never used".